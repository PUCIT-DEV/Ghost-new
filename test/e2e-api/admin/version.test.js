const {agentProvider, matchers} = require('../../utils/e2e-framework');
const {anyString, stringMatching} = matchers;

describe('API Versioning', function () {
    let agent;

    before(async function () {
        agent = await agentProvider.getAdminAPIAgent();
    });

    it('responds with no content version header when accept version header is NOT PRESENT', async function () {
        await agent
            .get('site/')
            .matchHeaderSnapshot({
                etag: anyString
            });
    });

    it('responds with current content version header when requested version is behind current version with no known changes', async function () {
        await agent
            .get('site/')
            .header('Accept-Version', 'v3.0')
            .matchHeaderSnapshot({
                etag: anyString,
                'content-version': stringMatching(/v\d+\.\d+/)
            });
    });
});
