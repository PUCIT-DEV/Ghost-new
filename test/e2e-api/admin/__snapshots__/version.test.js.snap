// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API Versioning responds with current content version header when requested version is behind current version with no known changes 1: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "167",
  "content-type": "application/json; charset=utf-8",
  "content-version": StringMatching /v\\\\d\\+\\\\\\.\\\\d\\+/,
  "etag": Any<String>,
  "vary": "Origin, Accept-Encoding",
  "x-powered-by": "Express",
}
`;

exports[`API Versioning responds with no content version header when accept version header is not present 1: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "167",
  "content-type": "application/json; charset=utf-8",
  "etag": Any<String>,
  "vary": "Origin, Accept-Encoding",
  "x-powered-by": "Express",
}
`;
